### PR TITLE
adds puffercarp

### PR DESCRIPTION
**Pull request by Screemonster**
Adds a cool and fun carp variant that looks a bit like a naval mine and acts a bit like a naval mine

Also fixes the failsnek appearing on mobs in vore.dmi when gibbed (an iconstate named "blank" is required for gib animations to work if a fallback icon is present in the file)

Puffercarps explode when exploded, explode as their attack, and explode when hit with thermal weapons - tasers _also_ cause burn damage to mobs so they also explode them. Brute damage does _not_ explode them, including brute weapons with an agony component such as beanbags, so ballistics are the way to go if you want to kill one without it blowing up.

### CODE REVIEW

**Comment by Skysteak** [comment]
1️⃣ 🐡 1️⃣ 
     
In all seriousness, seems like a fun concept, as carp variety is welcome in general.

File: code/modules/mob/living/simple_mob/subtypes/animal/space/carp.dm [hunk 1]
```diff
@@ -197,3 +197,53 @@
...
+		ready_to_blow = FALSE
+		gib()
+		var/turf/T = get_turf(src)
+		explosion(T, -1, -1, 4, 4)
+
+
+/mob/living/simple_mob/animal/space/carp/puffer/apply_melee_effects() //it gets close enough to attack? EXPLODE
+	kaboom()
+
+/mob/living/simple_mob/animal/space/carp/puffer/adjustFireLoss(var/amount) //you make it hot? EXPLODE
```
[the remark this answers is not in the document]
Comment (Screemonster) [reply code/modules/mob/living/simple_mob/subtypes/animal/space/carp.dm]: updated the thing, no idea why the compiler didn't complain

**Comment by KillianKirilenko** [comment]
New xenobio pet; deactivated puffercarp.

**Comment by Screemonster** [comment]
> New xenobio pet; deactivated puffercarp.

Entirely possible, just set ready_to_blow to false.
I thought about making fancy butchering code so that they'd go off if someone insufficently skilled tried to butcher them, or making it so you could strip meat off but if you didn't stop before the final step (which calls gib) it detonated, but I'd want to actually put a reward of some kind in first.

**Pull request merged**